### PR TITLE
[Wasm] Enable System.Data.Common tests

### PR DESCRIPTION
**Pull request by steveisok**
Disable the tests that require SQL Server.

### CODE REVIEW

File: src/libraries/System.Data.Common/tests/System/Data/Common/DbProviderFactoriesTests.cs [hunk 1]
```diff
@@ -69,6 +70,7 @@ public void GetFactoryWithDataRowTest()
         public void RegisterFactoryWithTypeNameTest()
         {
             ClearRegisteredFactories();
+            Console.WriteLine("Name: " + typeof(System.Data.SqlClient.SqlClientFactory).AssemblyQualifiedName);
```
Comment (mdh1418) [inline src/libraries/System.Data.Common/tests/System/Data/Common/DbProviderFactoriesTests.cs:73]: Remove? Assuming this was for debugging the test.

Comment (steveisok) [reply src/libraries/System.Data.Common/tests/System/Data/Common/DbProviderFactoriesTests.cs:73]: :-)

**Comment by MaximLipnin** [comment]
https://github.com/dotnet/runtime/pull/38963

**Comment by roji** [comment]
@steveisok is it possible to get a bit of context on why the DbProviderFactories and DataView APIs aren't supported on WASM?

**Comment by steveisok** [comment]
Not sure the official reason.  I was carrying it over from mono/mono.

I suspect the reason for `DbProviderFactories` may be to really safeguard you from running into a wall w/ SqlClient ;-).  That's just not going to work through a browser.

**Comment by ajcvickers** [comment]
@steveisok On the other hand, the SQLite ADO.NET provider already works on WASM, so any decisions here should not be based on whether or not the SQL Server ADO.NET provider will work. ADO.NET in general should work.

**Comment by steveisok** [comment]
@lewing Do you have any insight on this?

**Comment by lewing** [comment]
@ajcvickers we don't want to break things needlessly but we want to make sure the use experience is good, can you help with guidance here?

**Comment by roji** [comment]
In principle, I don't see a reason not to allow using DbProviderFactories, DataView or anything else that's under System.Data - these APIs seem to be implemented in pure .NET (no native dependencies or similar), and as @ajcvickers they can be used just like on .NET Core to work with Sqlite. It's possible that mono simply has/had a somewhat outdated implementation of DbProvidderFactories.

There are also various other database providers out there (e.g. Npgsql), which may or may not work with WASM. I don't think we need to do anything in System.Data itself to block anything - whether a provider works or not should be their issue.

**Comment by lewing** [comment]
Ok I think we'll roll back the PNSE and just disable the tests that require SQL Server.

File: src/libraries/System.Data.Common/tests/SuitePlatformDetection.cs [hunk 2]
```diff
@@ -0,0 +1,15 @@
...
+namespace System
+{
+    public static class SuitePlatformDetection
+    {
+        //
+        // Do not use the " { get; } = <expression> " pattern here. Having all the initialization happen in the type initializer
+        // means that one exception anywhere means all tests using PlatformDetection fail. If you feel a value is worth latching,
+        // do it in a way that failures don't cascade.
+        //
+        public static bool IsBinaryFormatterSupportedAndNotInvariantGlobalization => PlatformDetection.IsBinaryFormatterSupported && PlatformDetection.IsNotInvariantGlobalization;
```
Comment (akoeplinger) [inline src/libraries/System.Data.Common/tests/SuitePlatformDetection.cs:13]: This isn't needed.

You can just use the following instead, which is what we do that in a lot of places: 
```
[ConditionalFact(typeof(PlatformDetection), nameof(PlatformDetection. IsBinaryFormatterSupported), nameof(PlatformDetection.IsNotInvariantGlobalization)
```

**Review by akoeplinger — CHANGES_REQUESTED** [review]
Looks good in general, just one comment.

**Review by akoeplinger — APPROVED** [review]

**Pull request merged**